github-pr-background.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+/**
+ * JS that runs in the background and has access to browser tabs.
+ */
+
+browser.runtime.onMessage.addListener(createAttachTab);
+
+/**
+ * Retrieve the currently active tab.
+ */
+function getActiveTab(callback) {
+    var allTabsPromise = browser.tabs.query({
+        currentWindow: true
+    });
+    // FIXME(willkg): browser.tabs.query returns an array of *strings* for the
+    // tab ids. This makes no sense--seems like it should return an array of
+    // Tabs.
+    allTabsPromise.then((tabs) => {
+        for (let tabId in tabs) {
+            var isDone = false;
+            // Convert the string tabId to an int and get that tab
+            var getTabPromise = browser.tabs.get(parseInt(tabId));
+            getTabPromise.then((tab) => {
+                // If that tab is active, then call our callback.
+                if (tab.active) {
+                    callback(tab);
+                    isDone = true;
+                }
+            });
+
+            if (isDone) {
+                break;
+            }
+        }
+    });
+}
+
+
+/**
+ * Creates a new tab related to the currently active tab, open the
+ * Bugzilla attach url, and populate the fields.
+ */
+function createAttachTab(msg) {
```

Comment:
Converting this function to async/await instead of using Promises directly would improve readability.